### PR TITLE
feat(profiling): show update prompt if SDK updates are available

### PR DESCRIPTION
**Pull request by JonasBa**
CleanShot 2022-08-16 at 08 16 03@2x

### CODE REVIEW

**Comment by Jesse-Box** [comment]
@JonasBa can you think of a situation when it would be good to show this as a step to complete if the user has their SDK version updated?

**Comment by JonasBa** [comment]
@Jesse-Box I have missed your message. I dont know if this warrants a separate step because I would expect most folks to be close to the correct version here anyways

**Comment by Jesse-Box** [comment]
> @Jesse-Box I have missed your message. I dont know if this warrants a separate step because I would expect most folks to be close to the correct version here anyways

Ok cool, We're on the same page then! Based on your screenshot, I got the impression that the steps would show themselves, even though the person might have previously fulfilled that steps requirements. A good example of this would be the SDK version.

**Comment by JonasBa** [comment]
> > @Jesse-Box I have missed your message. I dont know if this warrants a separate step because I would expect most folks to be close to the correct version here anyways
> 
> Ok cool, We're on the same page then! Based on your screenshot, I got the impression that the steps would show themselves, even though the person might have previously fulfilled that steps requirements. A good example of this would be the SDK version.

Yeah, so actually now we always show the #2 step, but it shows if the SDK is on the correct version. It'll either say "you are using a good version of the sdk" (not exact text) or "you need to upgrade to v xxx"

I think it's ok that way tbh, it's still a good validation for the user to know that they are on a good versiob

File: static/app/components/profiling/ProfilingOnboarding/profilingOnboardingModal.tsx [hunk 1]
```diff
@@ -212,16 +225,73 @@ function SelectProjectStep({
...
+`;
+
+function AndroidInstallSteps({
+  project,
+  organization,
+}: {
+  organization: Organization;
+  project: Project;
+}) {
+  const sdkUpdates = useProjectSdkUpdates({organization, projectId: project.id});
```
Comment (Zylphrex) [inline static/app/components/profiling/ProfilingOnboarding/profilingOnboardingModal.tsx:284]: This fires a new request for each project select right? Should `useProjectSdkUpdates` only take the organization as a prop and let the per project check be done by the onboarding component?

File: static/app/components/profiling/ProfilingOnboarding/profilingOnboardingModal.tsx [hunk 2]
```diff
@@ -212,16 +225,73 @@ function SelectProjectStep({
...
+      </SDKUpdatesContainer>
+      {props.sdkUpdates.data === null ? (
+        <SdkUpdatesText>
+          {t('This project is using the latest SDK version.')}
+        </SdkUpdatesText>
+      ) : (
+        <SdkUpdatesText>
           {t(
-            'Make sure your SDKs are upgraded to at least version 6.0.0 (sentry-android).'
+            'For profiling to function, we requires you to update your Sentry SDK to version %s or higher.',
```
Comment (Zylphrex) [inline static/app/components/profiling/ProfilingOnboarding/profilingOnboardingModal.tsx:251]: This check that `sdkUpdates.data` is not null might be too restrictive since they may be on a version of the SDK that supports profiling already and still see this message when there's a newer version. Although we likely want them to update to the latest, this message seems to imply that their current version doesn't support profiling.

Comment (JonasBa) [reply static/app/components/profiling/ProfilingOnboarding/profilingOnboardingModal.tsx:251]: @Zylphrex Yes, that is true, but I dont know if we have any way to tell on which version the project is on exactly right now (or I missed it)

Comment (Zylphrex) [reply static/app/components/profiling/ProfilingOnboarding/profilingOnboardingModal.tsx:251]: unsure if we already have a way of parsing SDK version on the frontend but it looks like the SDK updates endpoint returns the current SDK version.

File: static/app/components/profiling/ProfilingOnboarding/profilingOnboardingModal.tsx (hunk 1 above)
Comment (JonasBa) [reply static/app/components/profiling/ProfilingOnboarding/profilingOnboardingModal.tsx:284]: @Zylphrex I can just make it load the data once and then just return resolve it

**Comment by Jesse-Box** [comment]
> I think it's ok that way tbh, it's still a good validation for the user to know that they are on a good versiob

So I understand where you're coming from, but we're leaving the possibility for confusion because the step says "Update" but the subtext below says "Latest". 

Bit of context, we recently had a bug in our dynamic sampling flow which has the same UX pattern of dynamically showing steps. This bug however, showed the SDK version even though the SDK version was latest and the user that reported the issue didn't know how to proceed.

Long story short, let's just hide this if the user has a compatible or latest SDK version.

**Review by Zylphrex — APPROVED** [review]

File: static/app/components/profiling/ProfilingOnboarding/profilingOnboardingModal.tsx [hunk 3]
```diff
@@ -12,8 +12,33 @@ import Tag from 'sentry/components/tag';
...
+ * Taken from https://gist.github.com/iwill/a83038623ba4fef6abb9efca87ae9ccb
+ * returns -1 for smaller, 0 for equals, and 1 for greater than
+ */
+function semverCompare(a: string, b: string): number {
+  if (a.startsWith(b + '-')) {
+    return -1;
+  }
+  if (b.startsWith(a + '-')) {
+    return 1;
+  }
```
[the remark this answers is not in the document]
Comment (JonasBa) [reply static/app/components/profiling/ProfilingOnboarding/profilingOnboardingModal.tsx:35]: @Zylphrex I just copied this from the comment, but I think this is to distinguish between 1.0.0 and 1.0.0-alpha case

[the remark this answers is not in the document]
Comment (Zylphrex) [reply static/app/components/profiling/ProfilingOnboarding/profilingOnboardingModal.tsx:35]: Can we add some tests for this function?

[the remark this answers is not in the document]
Comment (JonasBa) [reply static/app/components/profiling/ProfilingOnboarding/profilingOnboardingModal.tsx:35]: Sure, I'll just copy paste them from the gist though
